Add Cancel to Isabelle's Recovery

HisashiASM opened the pull request:
By pressing Guard, you will cancel out of uspecial, allowing Isabelle mains to recover

https://user-images.githubusercontent.com/98564957/171988042-c722efa9-390e-451b-95be-3614d8376b62.mp4

.

HisashiASM committed:
Add Cancel to Isabelle's Recovery

By pressing Guard, you will cancel out of uspecial, allowing Isabelle mains to recover.

HisashiASM committed:
Better handling of Cancel

3 Frame window before anything happens, Cancelable via Grab and Attack

HisashiASM committed:
Update opff.rs

Forgor to remove srings lol.

HisashiASM commented:
https://user-images.githubusercontent.com/98564957/172024758-de79b8ad-5ad8-4316-9260-c04ff42f1a47.mp4

Better handling Example

WuBoytH reviewed (dismissed):
This looks pretty solid, and a solid step from where it was yesterday.
LGTM

techyCoder81 commented:
all that remains now is for someone to test it in game to make sure theres no strange behavior

Riah321 commented:
This adds so much depth to the character it's unreal! Who knew that 10 lines of code could have such a huge impact on a character's gameplan. It all works as intended, and it fixes the exact issue it set out to fix while raising the skill ceiling a ton. After tweaks to the rest of her kit this character will be one of the funnest in HDR. Ship it!!!

MadVillainLordQ commented:
After testing in game, the change works as intended but I do have concerns over the balance of this style of change. First off, it seems the cancel can happen frame 1 of up B and you are only inactionable for 3 frames. This type of tool serves uses in neutral as Isabelle can essentially get two fairs/bairs out in one shorthop. While I do find this interesting, this does potentially change a lot of the balance with the character as it increases her offensive potential. I think recovery wise, it is extremely strong and basically transforms the characters recovery from being one of the weakest to one of the strongest of the game (if not just straight up the strongest). I do believe there should be some more startup for the cancel to occur along with longer inactionable frames after the cancel.

Discord link to video showcasing above:
https://discord.com/channels/659964948365049887/950979417004466196/982936050005602334

I'd also suggest changing the cancel animation VFX from the aura to the party popper thing from fsmash. (If that's possible of course).

HisashiASM commented:
I can nerf the timing of actionability from 3 to 5f, but anywhere past 10 it might just be pointless.
Issue is, if you get bopped after using it as a short hop or the like, you are SCREWED. The fuel is pretty much zeroed after you cancel to give the incentive to not spam it, and if you are off stage your recovery went from the best to one of the worst. Been playing it for a while and although I haven't had much time with it, I don't see it really mixing of a lot of her neutral minus scoring another fair by skipping jumpsquat, but your punishment again is no fuel for a good while.

HisashiASM commented:
I was thinking of making the cancel sfx the pocket sfx as it makes sense being a circle.

Riah321 commented:
Imo it's fine. Worst case it's slightly over tuned and we nerf it before it gets into beta. I'd rather have a possibly over tuned character than an unusable one.

Discussion on this change:
https://discord.com/channels/659964948365049887/660222338347827214/982937596818776104

jobrien97 commented:
> I was thinking of making the cancel sfx the pocket sfx as it makes sense being a circle.

Having gfx/sfx on cancel is unnecessary here imo. pls remove them

HisashiASM committed:
Merge branch 'dev' into pr/637

HisashiASM committed:
Nerfed Time of Detach and Frames until Active

Upped frames from 3 -> 5
It cannot be instantly cancel, instead awaiting for the initial flap to complete

techyCoder81 commented on fighters/shizue/src/opff.rs:
```diff
@@ -49,6 +49,18 @@ unsafe fn lloid_trap_fire_jc(boma: &mut BattleObjectModuleAccessor, status_kind:
+unsafe fn balloon_cancel(fighter: &mut L2CFighterCommon) {
+    if fighter.is_status_one_of(&[*FIGHTER_MURABITO_STATUS_KIND_SPECIAL_HI_WAIT, *FIGHTER_MURABITO_STATUS_KIND_SPECIAL_HI_FLAP]) {
+        // Cancel balloon trip early if character is holding shield, allowing for movement
+        println!("Is uspecial!");
```
please comment out or remove this println

techyCoder81 reviewed (dismissed):
almost there

HisashiASM committed:
Lol, I suck at removing Debug Strings

KekW

Riah321 commented:
Works. Merge it :0 (fixed the after airdodge glide toss)

I commented on the wrong PR lmao.

Riah321 commented:
Up special causes you to be unactionable for 22 frames. Is this intended behavior?
